lib/modules/datasource/terraform-provider/index.spec.ts

```diff
@@ -411,4 +414,86 @@ describe('modules/datasource/terraform-provider/index', () => {
       expect(res).toBeNull();
     });
   });
+
+  describe('getZipHashes', () => {
+    it('can fetch zip hashes', async () => {
+      httpMock
+        .scope(secondaryUrl)
+        .get(
+          '/terraform-provider-azurerm/2.56.0/terraform-provider-azurerm_2.56.0_SHA256SUMS'
+        )
+        .reply(200, azurerm2560Sha256Sums);
+
+      const res = await terraformProviderDatasource.getZipHashes([
+        {
+          name: 'azurerm',
+          version: '2.56.0',
+          os: 'linux',
+          arch: 'amd64',
+          filename: 'terraform-provider-azurerm_2.56.0_linux_amd64.zip',
+          url: 'https://releases.hashicorp.com/terraform-provider-azurerm/2.56.0/terraform-provider-azurerm_2.56.0_linux_amd64.zip',
+          shasums_url:
+            'https://releases.hashicorp.com/terraform-provider-azurerm/2.56.0/terraform-provider-azurerm_2.56.0_SHA256SUMS',
+        },
+      ]);
+
+      expect(res).not.toBeNull();
+      expect(res).toBeArray();
+      expect(res).toMatchObject([
+        '500d4e787bf046bbe64c4853530aff3dfddee2fdbff0087d7b1e7a8c24388628',
+        '8a7a6548a383a12aa137b0441c15fc7243a1d3e4fd8a9292946ef423d2d8bcff',
+        'd10b33dd19316ef10965ad0fb8ca6f2743bceaf5167bd8e6e25815e20786f190',
+        '39ff556080515b170b10f365a0f95abf2590e9ca3d79261defea1e3133e79088',
+        'ce72eaaecccb50f52f50c69ed3261b0a4050b846f2e664d120d30dfeb65067bc',
+        '1994185185046df38eb1d1ad3c3b07e4f964224e4ab756957473b754f6aec75c',
+        '766ff42596d643f9945b3aab2e83e306fe77c3020a5196366bbbb77eeea13b71',
+        'bb9f5e9289df17a7a07bdd3add79e41a195e3d129c2ab974b5bb6272c9812068',
+        '202556c142f001830dd4514d475dc747f863ad588382c43daa604d53761f59f5',
+        '3010bcf9ebe33e1195f0a7507183959918c6b88bbdc84c8cc96919654e0abcb0',
+        'fe5aba92430104238f66aaaf02acf323d457d387cd33d6b3d8c6fdd9e449b834',
+      ]);
+    });
+
+    it('does not fetch anything when there is no shasums_url defined', async () => {
+      const res = await terraformProviderDatasource.getZipHashes([
+        {
+          name: 'azurerm',
+          version: '2.56.0',
+          os: 'linux',
+          arch: 'amd64',
+          filename: 'terraform-provider-azurerm_2.56.0_linux_amd64.zip',
+          url: 'https://releases.hashicorp.com/terraform-provider-azurerm/2.56.0/terraform-provider-azurerm_2.56.0_linux_amd64.zip',
+          shasums_url: null,
```

Comment:
Right now there are: when data is fetched from releases.hashicorp.com, this endpoint is called for build information: https://releases.hashicorp.com/terraform-provider-google/4.84.0/index.json

So I have to keep the field in the interface nullable. In which case I would like to have a test for that case.

The required shasums url data is in the `shasums` fields, but it is not in any build information that is eventually passed to this method. I can map that data to each build by changing this code slightly, but it would still require the field in the interface to remain nullable, which probably still makes it a good idea to test that it doesn't hard-crash.
https://github.com/renovatebot/renovate/blob/4288293eec04a40ab9ef4662cc6b941a26012a21/lib/modules/datasource/terraform-provider/index.ts#L290-L299